client/components/PubHeader/PubHeader.js

```diff
@@ -57,19 +57,25 @@ class PubHeader extends Component {
 	}
 
 	componentWillUnmount() {
-		this.stickyInstance.cleanup();
+		this.cleanupSticky();
 		window.removeEventListener('resize', this.handleResize);
 	}
 
 	recalculateStickyOffset() {
 		const nextOffsetHeight = this.headerRef.current.offsetHeight;
 		if (nextOffsetHeight !== this.offsetHeight) {
 			this.offsetHeight = nextOffsetHeight;
-			this.stickyInstance.cleanup();
+			this.cleanupSticky();
 			this.stickyInstance = stickybits('.pub-header-component', { stickyBitStickyOffset: 35 - this.offsetHeight, useStickyClasses: true });
 		}
 	}
 
+	cleanupSticky() {
```

Comment:
For simple existence bugs like this, I tend to fix them by setting a default rather than a check. So, in the constructor, instead of `this.stickyInstance = undefined`, we could simply do `this.stickyInstance = { cleanup: ()=>{} }`. Obviously for more complex or deeply-nested existence issues, a check is simpler. In this case though, the one-line of default-setting feels simpler to reason about to me. Do you have thoughts or best-practice insight otherwise? (In the early days, I think there will be a lot of figuring out what is my personal quirk, and what is actually a good technical idea - so always feel free to give counter arguments!)

Also, there is a bit of storybook weirdness going on. I don't think what you did here actually solved the bug - but it did solve the problem. In storybook, pubHeader does indeed render with these changes - but the bug itself seems to come from pubDraftHeader unmounting.

<img width="509" alt="screen shot 2019-02-20 at 8 08 58 am" src="https://user-images.githubusercontent.com/1000455/53095370-a662df80-34ea-11e9-809b-1cab72ab47b4.png">

The bug is actually a typo in pubDraftHeader.js. `.cleanUp()` should be `.cleanup()`.

One last note on this: if we did use the `cleanupSticky()` function, we would have to initialize that from within the constructor: `this.cleanupSticky = this.cleanupSticky.bind(this)`. Initializing these functions is a bit verbose - but last I read up on it, the alternatives felt messier and more easily bug-ridden. Again - open to feedback on alternatives though.

Reply:
>  I pushed up [a commit](https://github.com/pubpub/pubpub/pull/306/commits/e3fffd2096f72b62c0cb175ab2de7608f970d9dd) with some linting and a change to pubDraftHeader - feel free to take a look at that.

That seems good to me. Sorry about the `cleanup`/`cleanUp` confusion! A few thoughts I have based on your comments:

- It would be nice to make this kind of bug impossible by using a static typechecker — either Flow or Typescript.
- I'm ±0 on the idea of default dummy objects like `{cleanup: () => {}}`; I find them helpful, but often difficult to maintain in the long term!
- I don't think it should be necessary to manually `bind` a class method as described unless you're planning on calling it outside of the class, e.g. by passing it as a callback prop to a React element in `render`.

Reply:
As an aside, it looks like certain code in `pubpub` is already using the [`@babel/plugin-proposal-class-properties`](https://babeljs.io/docs/en/next/babel-plugin-proposal-class-properties.html) transformer. In that case, I think we can start using arrow functions to replace the use of `bind` entirely, i.e. instead of:

```
class Test extends React.Component {
    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick() {
        console.log("oh no, I am clicked");
    }

    render() {
        return <button onClick={this.handleClick} />;
    }
}
```

we could use arrow functions to represent event handlers:

```
class Test extends React.Component {
    handleClick = () => {
        console.log("oh no, I am clicked");
    }

    render() {
        return <button onClick={this.handleClick} />;
    }
}
```

since arrow functions use lexical scoping for `this`, it will always refer to the class instance and we can avoid doing the binding. Last I checked (this morning) this feature isn't standard ES, but it's pretty far down the standards track and I feel pretty comfortable making use of it.

Reply:
Right - you only need to bind in the constructor if you're going to be calling `this` within the function and want it to refer to the Component scope. 

We used to use arrow functions as you described above, but we switched to constructor binding because:
1) I thought the arrow functions looked a bit ugly in component functions, especially when they had props
```
class Test extends React.Component {
    handleClick = (event, index, scope) => {
        console.log("oh no, I am clicked");
    }
}
```
In other contexts, I'm 100% on the arrow function train and our linter even enforces them over `function() {`.
2) something felt nice about having all the functions declared in the constructor as a means of communicating what functions exist within a component. This is less of an issue for small, simple components - but I do find nice with larger components like the Pub.js container.

Both these reasons feel a bit hand wavy - so I'm not against moving towards arrow binding for component functions. Let's continue offline - errr.... online on Slack :) - since I'm going to merge this PR now.